File: liqp-core/src/main/java/liqp/nodes/AndNode.java
package liqp.nodes;

import liqp.node.LNode;
import lombok.Getter;

@Getter
public class AndNode extends ExpressionNode {

  public AndNode(LNode lhs, LNode rhs) {
    super(lhs, rhs);
  }

  @Override
  public Object render(LContext context) {
    return context.isTrue(lhs.render(context)) && context.isTrue(rhs.render(context));
  }
}

File: liqp-core/src/main/kotlin/liqp/nodes/AndNode.kt
package liqp.nodes

import liqp.context.LContext
import liqp.node.LNode
import lombok.Getter

class AndNode(lhs: LNode, rhs: LNode) : ExpressionNode(lhs, rhs) {

  override fun render(context: LContext): Any? {
    return context.isTrue(lhs.render(context))
        && context.isTrue(rhs.render(context))
  }
}
